stubs/WTForms/wtforms/validators.pyi

```diff
@@ -0,0 +1,158 @@
+from collections.abc import Callable, Collection, Iterable
+from decimal import Decimal
+from re import Match, Pattern
+from typing import Any, TypeVar, overload
+
+from wtforms.fields import Field, StringField
+from wtforms.form import BaseForm
+
+_ValuesT = TypeVar("_ValuesT", bound=Collection[Any], contravariant=True)
+
+class ValidationError(ValueError):
+    def __init__(self, message: str = "", *args: object) -> None: ...
+
+class StopValidation(Exception):
+    def __init__(self, message: str = "", *args: object) -> None: ...
+
+class EqualTo:
+    fieldname: str
+    message: str | None
+    def __init__(self, fieldname: str, message: str | None = None) -> None: ...
+    def __call__(self, form: BaseForm, field: Field) -> None: ...
+
+class Length:
+    min: int
+    max: int
+    message: str | None
+    field_flags: dict[str, Any]
+    def __init__(self, min: int = -1, max: int = -1, message: str | None = None) -> None: ...
+    def __call__(self, form: BaseForm, field: StringField) -> None: ...
+
+class NumberRange:
+    min: float | Decimal | None
+    max: float | Decimal | None
+    message: str | None
+    field_flags: dict[str, Any]
+    def __init__(
+        self, min: float | Decimal | None = None, max: float | Decimal | None = None, message: str | None = None
+    ) -> None: ...
+    # any numeric field will work, for now we don't try to use a union
+    # to restrict to the defined numeric fields, since user-defined fields
+    # will likely not use a common base class, just like the existing
+    # numeric fields
+    def __call__(self, form: BaseForm, field: Field) -> None: ...
+
+class Optional:
+    string_check: Callable[[object], object]
```

Comment:
Seems like it's only expected to work with strs? And the return value is used as a boolean.
```suggestion
    string_check: Callable[[str], bool]
```